templates/common/on-prem/files/keepalived-script-default-ingress.yaml

```diff
@@ -3,4 +3,4 @@ path: "/etc/kubernetes/static-pod-resources/keepalived/scripts/chk_default_ingre
 contents:
   inline: |
     #!/bin/bash
-    /host/bin/oc --kubeconfig /var/lib/kubelet/kubeconfig get ep -n openshift-ingress router-internal-default -o yaml | grep -q 'ip: {{`{{.NonVirtualIP}}`}}$'
+    /host/bin/crictl -r unix:///host/run/crio/crio.sock pods --namespace openshift-ingress --state Ready | grep -q router-default
```

Comment:
In principle this looks fine to me. The one small concern I have is the grep - we've had enough cases of false positive matches when looking for patterns that I would like to limit this further if we can.

For example, what happens if someone shards their ingress and names theirs default-mycompany. Do we end up with an ingress pod named router-default-mycompany-blah-blah? This is not a rhetorical question, I honestly don't know what the behavior is and I would just like to verify that there's no possibility of overlapping names here, or if there is to find some way to handle that in the grep pattern.

Reply:
This is sadly a valid concern, by creating an IngressController named `default-chocobomb` we get the following pod

```
# oc get pod -n openshift-ingress -owide
NAME                                        READY   STATUS    RESTARTS   AGE     IP               NODE       NOMINATED NODE   READINESS GATES
router-default-65c4648b86-n5h4h             1/1     Running   0          15d     192.168.111.25   worker-2   <none>           <none>
router-default-65c4648b86-qz7jd             1/1     Running   0          15d     192.168.111.24   worker-1   <none>           <none>
router-default-chocobomb-775b66dc4d-6p82z   1/1     Running   0          2m34s   192.168.111.23   worker-0   <none>           <none>
```

so under the hood we end up with

```
[root@worker-0 ~]# crictl pods --namespace openshift-ingress --state Ready
POD ID              CREATED              STATE               NAME                                        NAMESPACE                  ATTEMPT             RUNTIME
5a4287444b4af       About a minute ago   Ready               router-default-chocobomb-775b66dc4d-6p82z   openshift-ingress          0                   (default)
81c4b257900ce       2 weeks ago          Ready               ingress-canary-t8nb4                        openshift-ingress-canary   0                   (default)
```

Reply:
We can try to be more specific with the grep, for example something like `grep 'router-default-[[:xdigit:]]\{10\}-[[:alnum:]]\{5\}'`. Would that work?